modules/porous_flow/src/bcs/PorousFlowEnthalpySink.C

```diff
@@ -0,0 +1,90 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowEnthalpySink.h"
+#include "Function.h"
+#include "PorousFlowDictator.h"
+#include "SinglePhaseFluidProperties.h"
+#include "MooseVariable.h"
+#include "PorousFlowSinkBC.h"
+
+#include "libmesh/quadrature.h"
+
+#include <iostream>
+
+registerMooseObject("PorousFlowApp", PorousFlowEnthalpySink);
+
+InputParameters
+PorousFlowEnthalpySink::validParams()
+{
+  InputParameters params = IntegratedBC::validParams();
+  params += PorousFlowSinkBC::validParamsCommon();
+  params.addRequiredParam<UserObjectName>(
+      "PorousFlowDictator", "The UserObject that holds the list of PorousFlow variable names");
+  params.addParam<unsigned int>(
+      "fluid_phase",
+      0,
+      "If supplied, then this BC will potentially be a function of fluid pressure.");
+  params.addClassDescription(
+      "Applies a source equal to the product of the mass flux and the "
+      "fluid enthalpy. The enthalpy is computed at temperature T_in and pressure equal to the "
+      "porepressure in the porous medium. Hence this adds heat energy "
+      "to the porous medium at rate corresponding to a fluid being injected at (porepressure, "
+      "T_in) at rate (-flux_function).");
+
+  return params;
+}
+
+PorousFlowEnthalpySink::PorousFlowEnthalpySink(const InputParameters & parameters)
+  : IntegratedBC(parameters),
+    _dictator(getUserObject<PorousFlowDictator>("PorousFlowDictator")),
+    _ph(getParam<unsigned int>("fluid_phase")),
```

Comment:
Need to check that `_ph` is less than the number of phases in the system (using `_dictator`).  And an exception test for this

Reply:
✅